runtime/src/accounts_index.rs

```diff
@@ -47,10 +46,19 @@ impl<T: Clone> AccountsIndex<T> {
 
     /// Get an account
     /// The latest account that appears in `ancestors` or `roots` is returned.
-    pub fn get(&self, pubkey: &Pubkey, ancestors: &HashMap<Fork, usize>) -> Option<(&T, Fork)> {
-        self.account_maps
-            .get(pubkey)
-            .and_then(|list| self.latest_fork(ancestors, list))
+    pub fn get(
+        &self,
+        pubkey: &Pubkey,
+        ancestors: &HashMap<Fork, usize>,
+    ) -> Option<(RwLockReadGuard<ForkList<T>>, usize)> {
```

Comment:
@aeyakovenko this is kind of ugly, but is there a better way?

Reply:
this is actually really nice. the only other approach is to pass a closure.